repository.go

```diff
@@ -16,12 +16,12 @@ type Repository interface {
 	EventsSince(time time.Time, limit int) ([]Event, error)
 }
 
-func unsavedEvents(objectEvents []Event, alreadySavedObjectEvents []Event) []Event {
-	knownIDs := make(map[string]interface{}, len(alreadySavedObjectEvents))
+func unsavedEvents(objectEvents []Event, knownEventIDs []string) []Event {
+	knownIDs := make(map[string]interface{}, len(knownEventIDs))
```

Comment:
```suggestion
	knownIDs := make(map[string]struct{}, len(knownEventIDs))
```

Légère optimisation que Guillaume m'avait déjà suggéré
https://go.dev/play/p/BGEYGOy9tPp